"ld: 7 duplicate symbols for architecture x86_64"

# Bug

Build fails when I try run `react-native run-ios`, this started to happen after I ran `pod install`

duplicate symbol _OBJC_IVAR_$_RCTSegmentedControl._selectedIndex in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControl.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControl.o)
duplicate symbol _OBJC_IVAR_$_RCTSegmentedControl._values in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControl.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControl.o)
duplicate symbol _OBJC_CLASS_$_RCTSegmentedControl in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControl.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControl.o)
duplicate symbol _OBJC_METACLASS_$_RCTSegmentedControl in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControl.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControl.o)
duplicate symbol _OBJC_IVAR_$_RCTSegmentedControl._onChange in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControl.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControl.o)
duplicate symbol _OBJC_CLASS_$_RCTSegmentedControlManager in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControlManager.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControlManager.o)
duplicate symbol _OBJC_METACLASS_$_RCTSegmentedControlManager in:
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/React-Core/libReact-Core.a(RCTSegmentedControlManager.o)
    /Users/Grape/xcode/bolan/ios/build/AwesomeProject/Build/Products/Debug-iphonesimulator/react-native-segmented-control/libreact-native-segmented-control.a(RCTSegmentedControlManager.o)
ld: 7 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


** BUILD FAILED **

## Environment info

System:
    OS: macOS 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Memory: 71.26 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.10.0 - /usr/local/bin/node
    npm: 6.10.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 23, 28
      Build Tools: 28.0.3
      System Images: android-28 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.3 AI-182.5107.16.33.5199772
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.5 => 0.60.5
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-git-upgrade: 0.2.7

Library version: 1.0.2

--- a/ios/RNCSegmentedControl.h
+++ b/ios/RNCSegmentedControl.h
@@ -9,7 +9,7 @@
 
 #import <React/RCTComponent.h>
 
-@interface RCTSegmentedControl : UISegmentedControl
+@interface RNCSegmentedControl : UISegmentedControl
 
 @property (nonatomic, copy) NSArray<NSString *> *values;
 @property (nonatomic, assign) NSInteger selectedIndex;

--- a/ios/RNCSegmentedControl.m
+++ b/ios/RNCSegmentedControl.m
@@ -5,13 +5,13 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import "RCTSegmentedControl.h"
+#import "RNCSegmentedControl.h"
 
 #import "RCTConvert.h"
 #import "RCTEventDispatcher.h"
 #import "UIView+React.h"
 
-@implementation RCTSegmentedControl
+@implementation RNCSegmentedControl
 
 - (instancetype)initWithFrame:(CGRect)frame
 {

--- a/ios/RNCSegmentedControl.xcodeproj/project.pbxproj
+++ b/ios/RNCSegmentedControl.xcodeproj/project.pbxproj
@@ -7,7 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		B3E7B58A1CC2AC0600A0062D /* RCTSegmentedControl.m in Sources */ = {isa = PBXBuildFile; fileRef = B3E7B5891CC2AC0600A0062D /* RCTSegmentedControl.m */; };
+		B3E7B58A1CC2AC0600A0062D /* RNCSegmentedControl.m in Sources */ = {isa = PBXBuildFile; fileRef = B3E7B5891CC2AC0600A0062D /* RNCSegmentedControl.m */; };
 /* End PBXBuildFile section */
 
 /* Begin PBXCopyFilesBuildPhase section */
@@ -23,9 +23,9 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
-		134814201AA4EA6300B7C361 /* libRCTSegmentedControl.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; includeInIndex = 0; path = libRCTSegmentedControl.a; sourceTree = BUILT_PRODUCTS_DIR; };
-		B3E7B5881CC2AC0600A0062D /* RCTSegmentedControl.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = RCTSegmentedControl.h; sourceTree = "<group>"; };
-		B3E7B5891CC2AC0600A0062D /* RCTSegmentedControl.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = RCTSegmentedControl.m; sourceTree = "<group>"; };
+		134814201AA4EA6300B7C361 /* libRNCSegmentedControl.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; includeInIndex = 0; path = libRNCSegmentedControl.a; sourceTree = BUILT_PRODUCTS_DIR; };
+		B3E7B5881CC2AC0600A0062D /* RNCSegmentedControl.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = RNCSegmentedControl.h; sourceTree = "<group>"; };
+		B3E7B5891CC2AC0600A0062D /* RNCSegmentedControl.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = RNCSegmentedControl.m; sourceTree = "<group>"; };
 /* End PBXFileReference section */
 
 /* Begin PBXFrameworksBuildPhase section */
@@ -42,7 +42,7 @@
 		134814211AA4EA7D00B7C361 /* Products */ = {
 			isa = PBXGroup;
 			children = (
-				134814201AA4EA6300B7C361 /* libRCTSegmentedControl.a */,
+				134814201AA4EA6300B7C361 /* libRNCSegmentedControl.a */,
 			);
 			name = Products;
 			sourceTree = "<group>";
@@ -50,8 +50,8 @@
 		58B511D21A9E6C8500147676 = {
 			isa = PBXGroup;
 			children = (
-				B3E7B5881CC2AC0600A0062D /* RCTSegmentedControl.h */,
-				B3E7B5891CC2AC0600A0062D /* RCTSegmentedControl.m */,
+				B3E7B5881CC2AC0600A0062D /* RNCSegmentedControl.h */,
+				B3E7B5891CC2AC0600A0062D /* RNCSegmentedControl.m */,
 				134814211AA4EA7D00B7C361 /* Products */,
 			);
 			sourceTree = "<group>";
@@ -59,9 +59,9 @@
 /* End PBXGroup section */
 
 /* Begin PBXNativeTarget section */
-		58B511DA1A9E6C8500147676 /* RCTSegmentedControl */ = {
+		58B511DA1A9E6C8500147676 /* RNCSegmentedControl */ = {
 			isa = PBXNativeTarget;
-			buildConfigurationList = 58B511EF1A9E6C8500147676 /* Build configuration list for PBXNativeTarget "RCTSegmentedControl" */;
+			buildConfigurationList = 58B511EF1A9E6C8500147676 /* Build configuration list for PBXNativeTarget "RNCSegmentedControl" */;
 			buildPhases = (
 				58B511D71A9E6C8500147676 /* Sources */,
 				58B511D81A9E6C8500147676 /* Frameworks */,
@@ -71,9 +71,9 @@
 			);
 			dependencies = (
 			);
-			name = RCTSegmentedControl;
+			name = RNCSegmentedControl;
 			productName = RCTDataManager;
-			productReference = 134814201AA4EA6300B7C361 /* libRCTSegmentedControl.a */;
+			productReference = 134814201AA4EA6300B7C361 /* libRNCSegmentedControl.a */;
 			productType = "com.apple.product-type.library.static";
 		};
 /* End PBXNativeTarget section */
@@ -90,7 +90,7 @@
 					};
 				};
 			};
-			buildConfigurationList = 58B511D61A9E6C8500147676 /* Build configuration list for PBXProject "RCTSegmentedControl" */;
+			buildConfigurationList = 58B511D61A9E6C8500147676 /* Build configuration list for PBXProject "RNCSegmentedControl" */;
 			compatibilityVersion = "Xcode 3.2";
 			developmentRegion = English;
 			hasScannedForEncodings = 0;
@@ -102,7 +102,7 @@
 			projectDirPath = "";
 			projectRoot = "";
 			targets = (
-				58B511DA1A9E6C8500147676 /* RCTSegmentedControl */,
+				58B511DA1A9E6C8500147676 /* RNCSegmentedControl */,
 			);
 		};
 /* End PBXProject section */
@@ -112,7 +112,7 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				B3E7B58A1CC2AC0600A0062D /* RCTSegmentedControl.m in Sources */,
+				B3E7B58A1CC2AC0600A0062D /* RNCSegmentedControl.m in Sources */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
@@ -235,7 +235,7 @@
 /* End XCBuildConfiguration section */
 
 /* Begin XCConfigurationList section */
-		58B511D61A9E6C8500147676 /* Build configuration list for PBXProject "RCTSegmentedControl" */ = {
+		58B511D61A9E6C8500147676 /* Build configuration list for PBXProject "RNCSegmentedControl" */ = {
 			isa = XCConfigurationList;
 			buildConfigurations = (
 				58B511ED1A9E6C8500147676 /* Debug */,
@@ -244,7 +244,7 @@
 			defaultConfigurationIsVisible = 0;
 			defaultConfigurationName = Release;
 		};
-		58B511EF1A9E6C8500147676 /* Build configuration list for PBXNativeTarget "RCTSegmentedControl" */ = {
+		58B511EF1A9E6C8500147676 /* Build configuration list for PBXNativeTarget "RNCSegmentedControl" */ = {
 			isa = XCConfigurationList;
 			buildConfigurations = (
 				58B511F01A9E6C8500147676 /* Debug */,

--- a/ios/RNCSegmentedControlManager.h
+++ b/ios/RNCSegmentedControlManager.h
@@ -7,6 +7,6 @@
 
 #import <React/RCTViewManager.h>
 
-@interface RCTSegmentedControlManager : RCTViewManager
+@interface RNCSegmentedControlManager : RCTViewManager
 
 @end

--- a/ios/RNCSegmentedControlManager.m
+++ b/ios/RNCSegmentedControlManager.m
@@ -5,19 +5,19 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import "RCTSegmentedControlManager.h"
+#import "RNCSegmentedControlManager.h"
 
 #import "RCTBridge.h"
 #import "RCTConvert.h"
-#import "RCTSegmentedControl.h"
+#import "RNCSegmentedControl.h"
 
-@implementation RCTSegmentedControlManager
+@implementation RNCSegmentedControlManager
 
 RCT_EXPORT_MODULE()
 
 - (UIView *)view
 {
-  return [RCTSegmentedControl new];
+  return [RNCSegmentedControl new];
 }
 
 RCT_EXPORT_VIEW_PROPERTY(values, NSArray<NSString *>)

--- a/js/RNCSegmentedControlNativeComponent.js
+++ b/js/RNCSegmentedControlNativeComponent.js
@@ -75,5 +75,5 @@ type NativeSegmentedControlIOS = Class<
 >;
 
 module.exports = ((requireNativeComponent(
-  'RCTSegmentedControl',
+  'RNCSegmentedControl',
 ): any): NativeSegmentedControlIOS);

--- a/js/SegmentedControlIOS.ios.js
+++ b/js/SegmentedControlIOS.ios.js
@@ -16,7 +16,7 @@ import {StyleSheet} from 'react-native';
 import type {SyntheticEvent} from 'CoreEventTypes';
 import type {ViewProps} from 'ViewPropTypes';
 
-import RCTSegmentedControlNativeComponent from './RCTSegmentedControlNativeComponent';
+import RNCSegmentedControlNativeComponent from './RNCSegmentedControlNativeComponent';
 
 type Event = SyntheticEvent<
   $ReadOnly<{|
@@ -63,7 +63,7 @@ type SegmentedControlIOSProps = $ReadOnly<{|
 
 type Props = $ReadOnly<{|
   ...SegmentedControlIOSProps,
-  forwardedRef: ?React.Ref<typeof RCTSegmentedControlNativeComponent>,
+  forwardedRef: ?React.Ref<typeof RNCSegmentedControlNativeComponent>,
 |}>;
 
 /**
@@ -102,7 +102,7 @@ class SegmentedControlIOS extends React.Component<Props> {
   render() {
     const {forwardedRef, ...props} = this.props;
     return (
-      <RCTSegmentedControlNativeComponent
+      <RNCSegmentedControlNativeComponent
         {...props}
         ref={forwardedRef}
         style={[styles.segmentedControl, this.props.style]}
@@ -122,7 +122,7 @@ const styles = StyleSheet.create({
 const SegmentedControlIOSWithRef = React.forwardRef(
   (
     props: SegmentedControlIOSProps,
-    forwardedRef: ?React.Ref<typeof RCTSegmentedControlNativeComponent>,
+    forwardedRef: ?React.Ref<typeof RNCSegmentedControlNativeComponent>,
   ) => {
     return <SegmentedControlIOS {...props} forwardedRef={forwardedRef} />;
   },

